trikot-datasources/datasources-flow/src/commonMain/kotlin/com/mirego/trikot/datasources/flow/generic/GenericDiskDataSource.kt

```diff
@@ -0,0 +1,69 @@
+package com.mirego.trikot.datasources.flow.generic
+
+import com.mirego.trikot.datasources.flow.BaseExpiringExecutableFlowDataSource
+import com.mirego.trikot.datasources.flow.DataSourceDispatchers
+import com.mirego.trikot.datasources.flow.ExpiringFlowDataSourceRequest
+import com.mirego.trikot.datasources.flow.FlowDataSourceExpiringValue
+import com.mirego.trikot.datasources.flow.filesystem.NativeFileSystem
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.Json
+import okio.FileNotFoundException
+import okio.Path
+import okio.Path.Companion.toPath
+
+class GenericDiskDataSource<R : ExpiringFlowDataSourceRequest, T>(
```

Comment:
Why generic? I'd go with a shorter term like
FileSystemDataSource or something like that.